public/app/features/templating/query_variable.ts

```diff
@@ -76,13 +62,141 @@ export class QueryVariable implements QueryVariableModel, VariableActions {
     private timeSrv: TimeSrv
   ) {
     // copy model properties to this instance
-    assignModelProperties(this, model, this.defaults);
+    if (model.isTemporary) {
+      this.temporary = {} as QueryVariableModel;
```

Comment:
This is a `temporary` model => this is when user clicks new in the Editor

Reply:
if we implemented the solution to this UX issue we can get rid of the temporary handling: https://github.com/grafana/grafana/issues/20703  , worth doing that before merging this?

Reply:
My next PR will migrate the Editor to React so it will be removed then at least in theory

Reply:
Maybe after the next PR is done we could get a clearer picture if this is a valid way of migrating the template variables to Redux?